frontend/components/commons/sidebar/SidebarButton.vue

```diff
@@ -35,27 +34,26 @@ export default {
       type: String,
       required: true,
     },
-    type: {
+    buttonType: {
       type: String,
-      required: true,
       validator: (value) => {
         return ["Mode", "Metrics", "Refresh"].includes(value);
       },
     },
   },
+  computed: {
+    isActive() {
+      return this.activeView.includes(this.id);
+    },
+    sidebarButtonClass() {
+      return [this.buttonType?.toLowerCase(), this.isActive ? "active" : null];
```

Comment:
code cleannning : 
1/ do one thing per line
2/ create constante or different computed for the items of the array